src/unity/python/turicreate/toolkits/object_detector/_sframe_loader.py

```diff
@@ -79,6 +83,9 @@ def _next(self):
         row = self.sframe[row_index]
         image = row[self.feature_column].pixel_data
 
+        if image.shape[2]>3:
```

Comment:
@srikris I tested this locally using a couple of dataframes generated from RGB-only images, RGBA images and CYMK images to make sure the behavior was as expected. But it's a great suggestion to add a test in test_image_type. I'll do so.

@nickjong that's a great catch I didn't test for 1-channel images, I was only focusing on the 4-channel cases. I'm guessing we would run into the same error as we did for CMYK. Writing out a unit test would be a good way to ensure all cases run smoothly. 

I was saving the overhead for the call between Python and C++ in invoking _convert_image_to_raw but we can remove the if check since you're right, there is a no op if the current channels and resize channels are the same. (Or we could add image.shape[2]<3 in the if-condition).

Reply:
I think it's certainly reasonable to want to avoid the overhead of `_convert_image_to_raw`, when you know we already have 3-channel input. If you go this route, I do think it might be more readable to query the number of channels from the Turi image, not from the numpy array. `turicreate.Image` exposes a `channels` property.